fix date/time widget for field with time zones

3nids opened the pull request:
this adds a new format of date/time to use Qt::ISODate format
the issue is that QDate::fromString does not recognize the t letter for the timezone in format
this a hack since any field_format needs to be tested if it corresponds to QgsDateTimeFieldFormatter::DEFAULT_ISO_FORMAT and in such case replace by Qt::ISODate instead of a format string

I am not really convinced by this patch.
Would like someone to review @m-kuhn @nyalldawson ? 

https://issues.qgis.org/issues/16657

3nids committed:
fix date/time widget for field with time zones

this adds a new format of date/time to use Qt::ISODate format
the issue is that QDate::fromString does not recognize the t letter for the timezone in format
this a hack since any field_format needs to be tested if it corresponds to QgsDateTimeFieldFormatter::DEFAULT_ISO_FORMAT and in such case replace by Qt::ISODate instead of a format string

3nids committed:
fix search widget wrapper

3nids commented:
ready for review.
@haubourg would be great if you could jump in

m-kuhn commented on src/core/fieldformatter/qgsdatetimefieldformatter.cpp:
```diff
@@ -22,6 +22,7 @@
 const QString QgsDateTimeFieldFormatter::DEFAULT_DATE_FORMAT = QStringLiteral( "yyyy-MM-dd" );
 const QString QgsDateTimeFieldFormatter::DEFAULT_TIME_FORMAT = QStringLiteral( "HH:mm:ss" );
 const QString QgsDateTimeFieldFormatter::DEFAULT_DATETIME_FORMAT = QStringLiteral( "yyyy-MM-dd HH:mm:ss" );
+const QString QgsDateTimeFieldFormatter::DEFAULT_ISO_FORMAT = QStringLiteral( "yyyy-MM-dd HH:mm:ss+t" );
```
Is there no need for a separated `ISO_DATE` and `ISO_DATE_TIME`?

3nids committed:
add a bool in config to determine if it uses Qt::ISODate for field format

also fixes the fact that the display format of QDateTimeEdit only accepts string formats, so in case of Qt::ISODate use a default date/time with zmne zone

3nids committed:
rename static variables

since they're not really default values

3nids commented:
I added a bool in the config to know if the field format is ISO or not.
This should avoid string comparison in the code (it happens only once in the config) and has a better readability.
Also fix the issue of the display format of QDateTimeEdit being string only (no Qt::ISODate possible).

m-kuhn commented on src/gui/editorwidgets/qgsdatetimeeditconfig.cpp:
```diff
@@ -23,6 +23,13 @@ QgsDateTimeEditConfig::QgsDateTimeEditConfig( QgsVectorLayer *vl, int fieldIdx,
+  mFieldFormatComboBox->addItem( tr( "Time" ), QgsDateTimeFieldFormatter::TIME_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "Date time" ), QgsDateTimeFieldFormatter::DATETIME_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "ISO date time" ), QgsDateTimeFieldFormatter::QT_ISO_FORMAT );
+  mFieldFormatComboBox->addItem( tr( "Custom" ), "" );
```
`""` > `QString()`?
